fastestimator/pipeline.py

```diff
@@ -139,102 +139,111 @@ def get_per_epoch(ops: Iterable[Union[T, Scheduler[T]]], epoch: int) -> List[T]:
 
 
 class BasePipeline:
-    def get_modes(self) -> Set[str]:
-        raise NotImplementedError()
-
-    def get_global_batch_size(self, mode: str, epoch: int) -> int:
-        raise NotImplementedError()
-
-    def get_signature_epochs(self, mode: str) -> List[int]:
-        raise NotImplementedError()
-
-    def get_all_output_keys(self) -> Set[str]:
+    def __init__(self,
+                 train_data: Union[Dataset, DataLoader, Scheduler, tf.data.Dataset],
+                 eval_data: Union[Dataset, DataLoader, Scheduler, tf.data.Dataset, None]):
+        self.train_data = train_data
+        self.eval_data = eval_data
+
+    def get_batch_size(self, mode: str, epoch: int) -> int:
+        return None
+
+    def get_iterator(self, mode: str, epoch: int) -> Iterator:
+        itr = None
+        if mode == "train":
```

Comment:
this implementation is inconsistent with the BasePipeline init args - which are not all directly iterators